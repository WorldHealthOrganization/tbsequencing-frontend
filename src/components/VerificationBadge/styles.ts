import { css } from '@emotion/react';

export const verificationBadge = css({
  borderRadius: '20px', // TODO taken on eye, could not take this from design
  boxShadow: '0px 0px 4px #C9DDEA',
  fontSize: 12,
  marginLeft: 36,
  height: 36,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  paddingLeft: 12,
  paddingRight: 12,
});

export const questionMark = css({
  fontWeight: 700,
  display: 'flex',
  marginRight: 3,
});
