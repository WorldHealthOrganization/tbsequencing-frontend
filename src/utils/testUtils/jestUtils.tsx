import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
