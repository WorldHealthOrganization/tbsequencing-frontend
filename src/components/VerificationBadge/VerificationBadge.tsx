/** @jsxImportSource @emotion/react */
import React from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';

export interface IVerificationBadgeProps {
  label: string;
  textIcon?: string;
  style?: SerializedStyles;
} // TODO move to shared?

export const VerificationBadge = ({ label, style, textIcon }: IVerificationBadgeProps) => (
  <div css={[styles.verificationBadge, style]}>
    {textIcon && <span css={styles.questionMark}>{textIcon}</span>}
    {' '}
    {label}
  </div>
);

export default VerificationBadge;
