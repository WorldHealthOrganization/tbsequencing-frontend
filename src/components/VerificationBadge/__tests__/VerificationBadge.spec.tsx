import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../utils/testUtils/jestUtils';
import VerificationBadge from '../VerificationBadge';

describe('VerificationBadge component test', () => {
  test('renders VerificationBadge component without error', () => {
    render(
      <VerificationBadge />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <VerificationBadge />,
    );

    expect(tree).toMatchSnapshot();
  });
});
