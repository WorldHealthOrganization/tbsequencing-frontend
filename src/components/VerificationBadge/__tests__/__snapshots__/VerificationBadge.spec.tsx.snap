// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationBadge component test Should render correctly 1`] = `
<div
  className="css-u2azet-VerificationBadge"
>
   
</div>
`;
